packages/odata-common/src/request-builder/get-all-request-builder-base.ts

```diff
@@ -61,13 +61,48 @@ export abstract class GetAllRequestBuilderBase<
    * @param orderBy - OrderBy statements to order the response by.
    * @returns The request builder itself, to facilitate method chaining.
    */
-  orderBy(orderBy: Orderable<EntityT>[]): this;
-  orderBy(...orderBy: Orderable<EntityT>[]): this;
   orderBy(
-    first: undefined | Orderable<EntityT> | Orderable<EntityT>[],
-    ...rest: Orderable<EntityT>[]
+    orderBy: OrderableAndOrderableInput<
+      EntityT,
+      DeSerializersT,
+      EntityApi<EntityBase, DeSerializersT>
+    >[]
+  ): this;
+  orderBy(
+    ...orderBy: OrderableAndOrderableInput<
+      EntityT,
+      DeSerializersT,
+      EntityApi<EntityBase, DeSerializersT>
+    >[]
+  ): this;
+  orderBy(
+    first:
+      | undefined
+      | OrderableAndOrderableInput<
+          EntityT,
+          DeSerializersT,
+          EntityApi<EntityBase, DeSerializersT>
+        >
+      | OrderableAndOrderableInput<
+          EntityT,
+          DeSerializersT,
+          EntityApi<EntityBase, DeSerializersT>
+        >[],
+    ...rest: OrderableAndOrderableInput<
+      EntityT,
+      DeSerializersT,
+      EntityApi<EntityBase, DeSerializersT>
+    >[]
   ): this {
-    this.requestConfig.orderBy = transformVariadicArgumentToArray(first, rest);
+    this.requestConfig.orderBy = transformVariadicArgumentToArray(
+      first,
+      rest
+    ).map(entity => {
```

Comment:
[pp]I would not use the term `entity` since we use it also for other things. Perhaps use `element` or `orderabelOrOrderableInput`

Reply:
I changed the parameter name entity.